### PR TITLE
Add java solution to CSES flight routes + fix formatting error in another solution

### PR DESCRIPTION
**Pull request by nathangong**

### CODE REVIEW

File: solutions/cses-1682.mdx [hunk 1]
```diff
@@ -108,4 +110,85 @@ int main() {
...
+		ArrayList<ArrayList<Integer>> adj2 = new ArrayList<>();
+
+		for (int i = 0; i < n; i++) {
+			adj1.add(new ArrayList<>());
+			adj2.add(new ArrayList<>());
+		}
+
+		for (int i = 0; i < m; i++) {
+			int a = io.nextInt()-1;
+			int b = io.nextInt()-1;
```
Comment (maggie-j-liu) [inline solutions/cses-1682.mdx:141]:
```suggestion
			int a = io.nextInt() - 1;
			int b = io.nextInt() - 1;
```

File: solutions/cses-1682.mdx [hunk 2]
```diff
@@ -108,4 +110,85 @@ int main() {
...
+
+		// Run dfs to check if you can reach all other vertices from vertex 1
+		vis = new boolean[n];
+		dfs(0, adj1);
+		for (int i = 0; i < n; i++) {
+			// If some vertex wasn't visited, that means that we cannot reach
+			// all other vertices, so we return NO
+			if (!vis[i]) {
+				io.println("NO");
+				io.println(1 + " " + (i+1));
```
Comment (maggie-j-liu) [inline solutions/cses-1682.mdx:154]:
```suggestion
				io.println(1 + " " + (i + 1));
```

File: solutions/cses-1682.mdx [hunk 3]
```diff
@@ -108,4 +110,85 @@ int main() {
...
+
+		// Run dfs to check if you can reach vertex 1 from all other vertices
+		vis = new boolean[n];
+		dfs(0, adj2);
+		for(int i = 0; i < n; i++) {
+			// If some vertex wasn't visited, that means vertex 1 cannot be
+			// reached from all other vertices, so we return NO
+			if (!vis[i]) {
+				io.println("NO");
+				io.println((i+1) + " " + 1);
```
Comment (maggie-j-liu) [inline solutions/cses-1682.mdx:168]:
```suggestion
				io.println((i + 1) + " " + 1);
```

**Review by maggie-j-liu — CHANGES_REQUESTED** [review]

File: solutions/cses-1682.mdx [hunk 4]
```diff
@@ -35,9 +36,10 @@ $u \rightarrow 1 \rightarrow v$, which contradicts the statement that
 $\texttt{can[u][v]} = 0$. Thus, $\texttt{can[u][v]} = 1$ for all vertices
```
Comment (dongliu0426) [inline solutions/cses-1682.mdx:36]: use texttt{true}/texttt{false} here (for consistency)

File: solutions/cses-1682.mdx (hunk 2 above)
Comment (maggie-j-liu) [reply solutions/cses-1682.mdx:154]: i think you missed this? https://github.com/cpinitiative/usaco-guide/pull/1616/files#r641886491

Comment (nathangong) [reply solutions/cses-1682.mdx]: Whoops yeah I think I missed that one

Comment (nathangong) [reply solutions/cses-1682.mdx]: Should be committed now

**Review by maggie-j-liu — APPROVED** [review]

**Pull request merged**